src/server-extension/resolvers/ChannelsResolver/utils.ts

```diff
@@ -105,3 +106,95 @@ export function buildExtendedChannelsQuery(
 
   return listQuery
 }
+
+export function buildTopSellingChannelsQuery(
+  args: TopSellingChannelsArgs,
+  info: GraphQLResolveInfo,
+  ctx: Context
+) {
+  const roundedDate = new Date().setMinutes(0, 0, 0)
+
+  const tree = getResolveTree(info)
+
+  // Extract subsquid-supported Channel sql args
+  const sqlArgs = parseSqlArguments(model, 'Channel', {
+    ...args,
+    where: args.where?.channel, // only supported WHERE part
+  })
+
+  // Extract subsquid-supported Channel fields
+  const channelSubTree = tree.fieldsByTypeName.TopSellingChannelsResult.channel
+  const channelFields = parseAnyTree(model, 'Channel', info.schema, channelSubTree)
+
+  // Generate query using subsquid's ListQuery
+  const listQuery = new ListQuery(model, ctx.openreader.dialect, 'Channel', channelFields, sqlArgs)
+  let listQuerySql = listQuery.sql
+
+  // Count NFT sells from the auctions and buy now events and add them to the query
+  listQuerySql = extendClause(listQuerySql, 'SELECT', '"top_selling_channels"."amount"')
+
+  listQuerySql = extendClause(
+    listQuerySql,
+    'FROM',
+    ` 
+           INNER JOIN (
+              WITH nftBoughtEvents AS (
+                SELECT
+                  (data#>>'{price}')::bigint AS price,
+                  data#>>'{previousNftOwner,channel}' AS previous_owner_id,
+                  "timestamp" 
+                FROM 
+                  processor.event
```

Comment:
I think you shouldn't query from `processor.event` in this case, as the `processor` schema includes entities which are hidden from the public view, while the default schema (`public`) doesn't.

For example, if you exlude a video using Orion, all the nfts, auctions, bids and events associated with this video will become hidden in `public` schema, but will still be accessible via `processor` schema. You will therefore:
- Count censored video nft sales into the result (which may not be desired)
- More importantly: You will run into issues when trying to join, for example, `event.data->>'winningBid'` from the `bid` table, as the bid will not exist in the `public` schema.